### ISSUE TITLE
TODOs for initial release

### ISSUE THREAD

**Issue by PEMapModder** [LegendOfMCPE/PocketFactions#2]
- [x] Work with xEcon (dependency)
  - [x] xEcon permissions
- [x] Commands
  - [x] /f
    - [x] /f claim
      - [x] power check (needs to wait for mroe StatsCore functions)
      - [x] xEcon money check
    - [x] /f create
      - [x] name validation
        - [x] regular expression in config file
        - [x] avoid repetitions
    - [x] /f disband
    - [x] /f donate
    - [x] /f home
      - [x] multiple homes
    - [x] /f homes
    - [x] /f info
    - [x] /f invite
    - [x] /f join
    - [x] /f kick
    - [x] /f money
      - [x] deposit
      - [x] withdraw
      - [x] check
    - [x] /f motto
      - [x] take money
    - [x] /f perm
      - [x] assigning members to ranks
      - [x] adding new ranks
      - [x] renaming ranks
      - [x] default/truce/ally/standard ranks
      - [x] removing ranks and safely reassigning members to other ranks
    - [x] /f quit
    - [x] /f rel
    - [x] /f rename
      - [x] take money
    - [x] /f sethome
      - [x] multiple homes (wrote this again to spam TODOs to look nicer xD)
    - [x] /f setopen
      - [x] take money
    - [x] /f siege
    - [x] /f unclaim
    - [x] /f unclaimall
  - [x] /fadm
    - [x] /fadm on
    - [x] /fadm off (spamming checks to make it look nicer :P)
- [x] Work with the permissions internally and permissions externally (relations; with ref to MassiveCraft/Factions)
  - [x] Internal ranks
  - [x] External relations
    - [x] Allies
    - [x] Truce
    - [x] Neutral
    - [x] Enemy
  - [x] Building permissions
  - [x] PvP permissions
- [X] Add redirector from commands to functions
- [x] Power (dependency StatsCore will do it)
- [x] faster searching (refer to #7)
- [x] "Sieging" chunks (claiming a chunk that was claimed by a faction with overdraft chunks:power)
  - [x] Arguments for sieging chunks
    - [x] Owner faction has power overdraft
    - [x] Sieger faction has excess power even after sieging
    - [x] Close enough to the chunk centre to claim it
- [ ] Staff commands
- [ ] Debug!
- [ ] External API
  - [ ] Event system
  - [x] Convenient getters

**Comment by iJoshuaHD** [comment]
5 checklist that looks so simple to be finished xD

**Comment by lgibson02** [comment]
I personally hate using GitHub.

**Comment by iJoshuaHD** [comment]
BitBucket.org wont allow me to use my GitHub account which sucks :(

**Comment by PEMapModder** [comment]
This sounds so simple. After an hour reading all the PC  documentation I think I should add more details here.

**Comment by PEMapModder** [comment]
Eight checks now. How do you feel? Still looking so simple? XD

**Comment by iJoshuaHD** [comment]
it looks still simple but the work is getting harder xD

**Comment by PEMapModder** [comment]
10 now after the disccusion just now. (I talked with you for two hours! I didn't know that, @ijoshuahd!)

**Comment by iJoshuaHD** [comment]
now my nose starts to irretate xD almost bleeding :D

**Comment by PEMapModder** [comment]
lol

**Comment by iJoshuaHD** [comment]
i guess we should remove/cancel the seiging feature.

**Comment by PEMapModder** [comment]
Why?

The owner can disable the feature if he wants.

**Comment by iJoshuaHD** [comment]
oh well... it sucks because its annoying to see factions claimed beside your faction.

**Comment by PEMapModder** [comment]
Cleaned some duplicated checks with nests.
How does it look now?

**Comment by PEMapModder** [comment]
22/51 tasks complete now!

**Comment by iJoshuaHD** [comment]
have you think that if you donate xEcon Money to the faction money, they can claim more chunks? what i mean is, Can you remove the feature where faction members/owner could NOT add money to their Faction, instead, the faction will only earn money on certain reasons. example is when they kill another faction team players.

the reason why is, they could just make another player account, add them to the faction, that player account has a default of 1000$(for example) xEcon Money, that money will be donated to the faction and they'll do that process all over and over again.

**Comment by PEMapModder** [comment]
They need both money and power to claim chunks.

First of all, about xEcon, players can always pay themselves money by new accounts using /pay. Discuss about that in LegendOfMCPE/xEcon#1.
Secondly, they need power to claim chunks. All players start with 0 power. XEcon money only makes them to need to have one more thing to give when they claim chunks. But their total number of chunks is still controlled by their power such that if they don't have enough total online time, they can't claim chunks.
If they have time to make new accounts, they get higher total online time by staying at the same account.

As for the xEcon thing, they can be prevented by instead of giving each account default money, give each IP default money only once. And don't talk to me about VPN before you find me a Bukkit plugin that disallows using VPN networks to bypass IP ban.

**Comment by PEMapModder** [comment]
I'll stick to @massivecraft's plugin in the inter faction relations.

**Comment by PEMapModder** [comment]
Also, players can't withdraw money from the faction account.

**Comment by iJoshuaHD** [comment]
i mean deposit xEcon money. Dont allow players to deposit money to their factions. And, we should limit the number of claims to avoid large part of the world claimed by one faction if their powers is plenty.

**Comment by PEMapModder** [comment]
Why not? Power is also calculated via kills and deaths. Anyway, the maximum claims can be limited, of course, easily. Just add that. However, I personally don't advise that.

**Comment by PEMapModder** [comment]
Added three more TODOs at the bottom.

**Comment by PEMapModder** [comment]
55/66 checks now!

**Comment by lgibson02** [comment]
I think /f perm should be /f rank. I just think it goes better since it's for editing ranks.

**Comment by PEMapModder** [comment]
@DinoKill your choice :) as you like

And you don't need to modify the class name if you don't like to

**Comment by PEMapModder** [comment]
An idea:

```
/f rank appoint|app|promote|demote <player> <rank>
/f rank add|new <new rank name> <permission>,[permission],[permission]...
/f rank modify|mod <rank> <permission>,[permission]...
/f rank remove|rm <rank name> [rank to reassign to = default rank] (rank to assign to is what rank should players with this rank be reassigned to)
/f rank rename <old rank name> <new rank name> (the rank ID remains unchanged, so maybe add function $rank->setName())
/f rank set <rank> <default|truce|ally|standard>
```

It is just an idea I have a few minutes ago. You might think of, like and use a better method :)

Also, standard doesn't sound representable. Maybe you are English and think of better words xD

**Comment by iJoshuaHD** [comment]
those new features would clearly extend the release date thrice xD

**Comment by PEMapModder** [comment]
It is not a new feature. It is a planned feature :D

**Comment by iJoshuaHD** [comment]
planned feature would still be new feature once its done :P same lol xD

**Comment by PEMapModder** [comment]
Checked!

**Comment by PEMapModder** [comment]
@ijoshuahd Admin commands? No idea what to do. IRC?

**Comment by iJoshuaHD** [comment]
ok

**Comment by iJoshuaHD** [comment]
omg, 4 checklist to go :O

**Comment by PEMapModder** [comment]
@ijoshuahd you forgot xEcon and StatsCore :unamused:

**Comment by DovahDoVolom** [comment]
Will this work with Economies? Plugin by OneBone

**Comment by iJoshuaHD** [comment]
@DovahDoVolom i guess it only supports xEcon for now.

**Comment by PEMapModder** [comment]
xEcon is the only economy provider that is possible in our case, because the faction money is saved by the economy plugin, not by PocketFactions, and xEcon is the so far only plugin that allow non-player account owners.

**Comment by lgibson02** [comment]
Can't we add a feature where if xEcon is not present it saves the faction's balance in it's own file. No offense, but I don't think everyone will want to use xEcon.

**Comment by PEMapModder** [comment]
Would be a bit complicated at the current stage.

**Comment by PEMapModder** [comment]
I understand. Therefore, during release I'll try all possible measures to let people port economys database to xEcon.

**Comment by iJoshuaHD** [comment]
Events?

**Comment by PEMapModder** [comment]
@ijoshuahd an external events API

**Comment by PEMapModder** [comment]
I'll open another issue for the rewrite in PocketFactions Beta.

**Comment by iJoshuaHD** [comment]
how about xEcon integration?

**Comment by PEMapModder** [comment]
It should support any economy plugins.
